sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -689,8 +689,7 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
         than 1. If None, there is no maximum limit.
     max_depth : int or None, optional (default=None)
         The maximum depth of each tree. The depth of a tree is the number of
-        nodes to go from the root to the deepest leaf. Must be strictly greater
-        than 1. Depth isn't constrained by default.
+        nodes to go from the root to the deepest leaf. Depth isn't constrained by default.
```

Comment:
```suggestion
        edges to go from the root to the deepest leaf. Depth isn't constrained by default.
```

Reply:
Please also update the classifier's docstring

Reply:
@NicolasHug Thanks for the review. I have updated the docstrings and also added a test accordingly.